ark/mibi/tiling_utils.py

```diff
@@ -0,0 +1,240 @@
+                              fov['name']))
+
+        while randomize not in [0, 1]:
+            print("Error: randomize parameter must be 0 or 1")
+            randomize = int(input("Randomize? Enter 0 for no and 1 for yes: "))
+
+        region_rand.append(randomize)
+
+    # need to copy fov metadata over, needed for create_tiled_regions
+    tiling_params['fovs'] = copy.deepcopy(fov_tile_info['fovs'])
+
+    # assign fields to tiling_params
+    tiling_params['region_start_x'] = region_start_x
+    tiling_params['region_start_y'] = region_start_y
+    tiling_params['fov_num_x'] = fov_num_x
+    tiling_params['fov_num_y'] = fov_num_y
+    tiling_params['x_fov_size'] = x_fov_size
+    tiling_params['y_fov_size'] = y_fov_size
+    tiling_params['randomize'] = region_rand
+
+    moly_run_insert = int(
+        input("Insert moly points between runs? Enter 0 for no and 1 for yes: ")
+    )
+
+    while moly_run_insert not in [0, 1]:
+        print("Error: moly point run parameter must be either 0 or 1")
+        moly_run_insert = int(
+            input("Insert moly points between runs? Enter 0 for no and 1 for yes: ")
+        )
+
+    tiling_params['moly_run'] = moly_run_insert
+
+    # whether to insert moly points between tiles
+    # NOTE: moly points will be inserted between different runs regardless of what's set here
+    moly_interval_insert = int(
+        input("Specify moly point tile interval? Enter 0 for no and 1 for yes: ")
+    )
+
+    while moly_interval_insert not in [0, 1]:
+        print("Error: moly interval insertion parameter must enter 0 or 1")
+        moly_interval_insert = int(
+            input("Specify moly point tile interval? Enter 0 for no and 1 for yes: ")
+        )
+
+    # if moly insert is set, we need to specify an additional moly_interval param
+    # NOTE: the interval applies regardless of if the tiles overlap runs or not
+    if moly_interval_insert:
+        moly_interval = int(input("Enter the fov interval size to insert moly points: "))
+
+        while moly_interval < 1:
+            print("Error: moly interval must be positive")
+            moly_interval = int(input("Enter the fov interval size to insert moly points: "))
+
+        tiling_params['moly_interval'] = moly_interval
+
+    return tiling_params, moly_point
+
+
+def create_tiled_regions(tiling_params, moly_point):
+    """Create the tiled regions for each fov
+
+    Args:
+        tiling_params (dict):
+            The tiling parameters created by set_tiling_params
+        moly_point (dict):
+            The moly point to insert between fovs (and intervals if specified in tiling_params)
+
+    Returns:
+        dict:
+            Data containing information about each tile, will be saved to JSON
+    """
+
+    # helper function for creating all pairs between two lists
+    def pairs(*lists):
+        for t in combinations(lists, 2):
+            for pair in product(*t):
+                yield pair
```

Comment:
since this helper function doesn't depend on args/vars in the function scope, it probs should be moved outside the function